ctk/common/src/main/kotlin/org/codice/compliance/verification/core/SamlDefinedIdentifiersVerifier.kt

```diff
@@ -118,4 +130,48 @@ internal class SamlDefinedIdentifiersVerifier(val node: Node) {
             }
         }
     }
+
+    /** 8.3.7 Persistent Identifier */
+    private fun verifyPersistentIdentifiers() {
+        node.recursiveChildren()
+            .filter { it.namespaceURI != null }
+            .filter { it.namespaceURI == PERSISTENT_ID }
+            .forEach {
+                if (it.textContent != null && it.textContent.length > ID_VALUE_LENGTH_LIMIT)
+                    throw SAMLComplianceException.create(SAMLCore_8_3_7_a,
+                        message = "The length of the Persistent ID's value was greater than " +
+                            "$ID_VALUE_LENGTH_LIMIT characters.",
+                        node = it)
+
+                if (it.children("NameQualifier")
+                        .any { it.textContent != idpMetadata.entityId })
```

Comment:
❗️ What if the IdP we are testing got an assertion from another SAML party. Then the NameQualifier wouldn't be the EntityID from the metadata. See `SAMLCore_8_3_7_b` and `SAMLCore_8_3_7_c`.

Reply:
It says the entity id of the `identity provider`